ethcore/sync/src/chain/mod.rs

```diff
@@ -821,6 +852,21 @@ impl ChainSync {
 		self.active_peers = self.peers.keys().cloned().collect();
 	}
 
+	/// Add a request for later processing
+	pub fn add_delayed_request(&mut self, peer: PeerId, packet_id: u8, data: &[u8]) {
+		if let Some(position) = self.delayed_requests.iter().position(|request| request.0 == peer && request.1 == packet_id) {
```

Comment:
Would it be more efficient to use `binary_search_by()` here, to keep the vector sorted at all times?

Reply:
I've made it a vector due to @tomusdrw very good suggestion to keep the order of insertions. So sorting won't help (if i understand your suggestion correctly).

Reply:
Why is the deduplication needed anyway? Is `data` expected to be different? I'd just `push` unconditionally and rather restrict the size of `delayed_requests` to make sure we don't use up too much resources.